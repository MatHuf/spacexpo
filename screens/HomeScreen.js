import React from "react";
import { Image, Text, View, Button } from "react-native";
import { GET_HISTORY, GET_LAUNCHES } from "../queries";
import { HistoryListItem, LaunchListItem } from "../components/listItems";
import {
  launchesMap,
  LaunchSectionHeader
} from "../components/listItems/LaunchListItem";
import { homeScreenStyles as styles } from "../styles";
import StarlinkLaunch from "../assets/images/StarlinkLaunch.jpg";

export default function HomeScreen({ navigation }) {
  return (
    <View style={styles.container}>
      <Image source={StarlinkLaunch} style={styles.image} />
      <Text style={styles.title}>Welcome to SpaceXpo</Text>
      <Button
        title="Launches"
        onPress={() =>
          navigation.navigate("Launches", {
            query: GET_LAUNCHES,
            listKey: "launches",
            renderItem: LaunchListItem,
            itemKey: "flight_number",
            hasSections: true,
            renderSectionHeader: LaunchSectionHeader,
            resultMap: result => launchesMap(result)
          })
        }
      />
      <Button
        title="Vehicles"
        onPress={() => navigation.navigate("Vehicles")}
      />
      <Button
        title="History"
        onPress={() =>
          navigation.navigate("History", {
            query: GET_HISTORY,
            listKey: "history",
            renderItem: HistoryListItem,
            itemKey: "id"
          })
        }
      />
    </View>
  );
}

HomeScreen.navigationOptions = {
  headerShown: false
};
