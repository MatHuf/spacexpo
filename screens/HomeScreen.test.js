import React from "react";
import renderer from "react-test-renderer";

import HomeScreen from "./HomeScreen";

describe("<Home />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(React.createElement(HomeScreen)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
