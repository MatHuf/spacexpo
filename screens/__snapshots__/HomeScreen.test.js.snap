// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 600,
        "marginBottom": 16,
        "width": 400,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 32,
      }
    }
  >
    Welcome to SpaceXpo
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Launches
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Vehicles
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        History
      </Text>
    </View>
  </View>
</View>
`;
